src/main/java/uk/gov/hmcts/reform/cpo/service/impl/CasePaymentOrdersServiceImpl.java

```diff
@@ -111,6 +101,7 @@ public CasePaymentOrder updateCasePaymentOrder(UpdateCasePaymentOrderRequest req
         throw new UnsupportedOperationException("Implement me: see CPO-6");
     }
 
+    @Transactional(rollbackOn = CasePaymentOrderCouldNotBeFoundException.class)
```

Comment:
do we need rollbackOn param here? I guess spring automatically rollback if it encounters an exception.

Reply:
You are correct - Runtime exceptions cause rollback, but not checked exceptions.

This is now a Runtime Exception, but used to be a checked exception - will remove

👍

Reply:
Resolved in https://github.com/hmcts/cpo-case-payment-orders-api/pull/63/commits/08f287c51a39f85e5dc281a0d29ea6f351a6c432